tests/Doctrine/Tests/ORM/Functional/Ticket/GH11037/InvalidEntityWithTypedEnum.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket\GH11037;
+
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\Id;
+
+/**
+ * @Entity
+ */
+class InvalidEntityWithTypedEnum
+{
+    /**
+     * @Id
+     * @Column
+     */
+    protected int $id;
+
+    /**
+     * @Column(type="integer", enumType=StringEntityStatus::class)
+     */
+    protected StringEntityStatus $status1;
+
+    /**
+     * @Column(type="integer", enumType=StringEntityStatus::class)
+     */
+    protected IntEntityStatus $status2;
```

Comment:
Here the enum is not the same for the property and the mapping. Are there cases where that is not a problem? If not, then there is no point in looking at the backing type, is there?

Reply:
True, there is no point in checking the backing type if the `enumType` doesn't match the enum class at that moment. Conversely, there is no point in checking the `enumType` if the mapping `type` doesn't match the backing type.

Since this is an AND condition, it's necessary to validate both the mapping `type` and the `enumType` against the property type, so I needed to choose one of them to check first, and I opted to check the mapping `type` against the backing type first because, if it fails, it returns the default error message.

Please let me know if anything remains unclear or if you have a preferred code in mind.

Reply:
Thanks, I finally understand what I overlooked.